### DISCUSSION TITLE
Published changelog different from generated changelog ?

### DISCUSSION THREAD

**Discussion by sloonz** [goreleaser/goreleaser#2283] [Q&A] [answered]
Hello, [uback](https://github.com/sloonz/uback) here, trying to use goreleaser for my releases.

v0.3 consists of three commits, one of those being ignored for changelog generation : 

``` $git log v0.2..v0.3
commit 83702ae38f0f7145988c98c39a012642906f437e (HEAD -> main, tag: v0.3, origin/main)
Date:   Fri Jun 4 17:30:46 2021 +0200

    refactor!: use age for encryption
    
    BREAKING CHANGE: uback now uses age for file encryption, which allows us
    to avoid doing a lot of cryptographic stuff (and potentially mistakes)
    ourself. As a consequence, the file format is changed and is backups
    generated by previous versions of uback can no longer be restored by
    this release. Moreover, keys have their format changed too and new ones
    must be generated.

commit 009c847743d37ec7b8decd9b1ce1fb16d2d73367
Date:   Tue Jun 1 14:27:05 2021 +0200

    feat: allow custom sources and destinations commands to take multiple arguments

commit 9ab30e8c21509f71358ddc2ada5be5b43c4c2569
Date:   Mon May 31 10:31:57 2021 +0200

    build: skip more commits in changelog
```

If I run `goreleaser release --rm-dist --skip-publish` locally, `dist/CHANGELOG.md` is pretty much what I expected : 

```
## Changelog

009c847 feat: allow custom sources and destinations commands to take multiple arguments
83702ae refactor!: use age for encryption
```

However, only  83702ae appears in the github release : https://github.com/sloonz/uback/releases

What am I doing wrong ?

**Comment by caarlos0** [comment]
Hey @sloonz , seems to be because of this https://github.com/sloonz/uback/blob/main/.goreleaser.yml#L37

**Reply by sloonz** [reply]
That excludes 9ab30e8c21509f71358ddc2ada5be5b43c4c2569, but that was intentional. My question is why 009c847 `feat: allow custom sources and destinations commands to take multiple arguments`) doesn't appear.

**Reply by caarlos0** [reply]
oh... that seems like a bug 🤔 , will try to debug

**Reply by caarlos0** [reply]
oh, nevermind, you're running against a shallow clone: https://github.com/sloonz/uback/runs/2747428021?check_suite_focus=true#step:4:20

**Reply by caarlos0** [reply]
This reminds me I need to work on a FAQ, this happens every once in a while and even I forget to check that sometimes 🤔

**Comment by caarlos0** [comment, marked as answer]
Running against a [shallow clone](https://github.com/sloonz/uback/runs/2747428021?check_suite_focus=true#step:4:20), GoReleaser needs a full clone to work.

Check https://goreleaser.com/ci/actions/ for mor info.

**Reply by sloonz** [reply]
That was my issue, thanks !
